@here/harp-features-datasource/lib/FeaturesDataSource.ts

```diff
@@ -0,0 +1,166 @@
+        });
+    }
+
+    /**
+     * Creates a polygon feature.
+     *
+     * @param coordinates The geometry of the feature.
+     * @param style The style to draw the feature with.
+     */
+    createPolygon(coordinates: Polygon["coordinates"], properties?: {}): MapViewPolygonFeature {
+        return new MapViewPolygonFeature(coordinates, properties);
+    }
+
+    /**
+     * Creates a line feature.
+     *
+     * @param coordinates The geometry of the feature.
+     * @param style The style to draw the feature with.
+     */
+    createLine(coordinates: LineString["coordinates"], properties?: {}): MapViewLineFeature {
+        return new MapViewLineFeature(coordinates, properties);
+    }
+
+    /**
+     * Creates a point feature.
+     *
+     * @param coordinates The geometry of the feature.
+     * @param style The style to draw the feature with.
+     */
+    createPoint(coordinates: Point["coordinates"], properties?: {}): MapViewPointFeature {
+        return new MapViewPointFeature(coordinates, properties);
+    }
+
+    /**
+     * Adds a custom feature in the datasource.
+     *
+     * @param features
+     */
+    add(...features: MapViewFeature[]) {
+        for (const feature of features) {
+            this.addFeature(feature);
+        }
+        this.update();
+    }
+
+    /**
+     * Removes a custom feature in the datasource.
+     *
+     * @param feature
+     */
+    remove(...features: MapViewFeature[]) {
+        for (const feature of features) {
+            this.removeFeature(feature);
+        }
+        this.update();
+    }
+
+    /**
+     * Removes all the custom features in this `FeaturesDataSource`.
+     */
+    clear() {
+        this.m_featureCollection = {
+            type: "FeatureCollection",
+            features: []
+        };
+        this.update();
+    }
+
+    private addFeature(feature: MapViewFeature) {
+        // Check if the feature is not already in there.
+        for (const _feature of this.m_featureCollection.features) {
+            if (
+                _feature.properties !== undefined &&
+                _feature.properties.__mapViewUuid === feature.uuid
+            ) {
+                return;
+            }
+        }
+
+        // Create a GeoJson feature from the feature coordinates and push it.
+        const geometry: FeatureGeometry = {
+            type: feature.type,
+            coordinates: feature.coordinates
+        } as any;
+        const geojsonFeature: Feature = {
+            type: "Feature",
+            geometry,
+            properties: {
+                ...feature.properties,
+                __mapViewUuid: feature.uuid
+            }
+        };
+        this.m_featureCollection.features.push(geojsonFeature);
+    }
+
+    private removeFeature(feature: MapViewFeature) {
+        // Remove geojson feature from the root FeatureCollection.
+        const features = this.m_featureCollection.features;
```

Comment:
why not use features.find((value,index, obj) => {
...?
});

Reply:
We need the index to then splice the array, so did as you suggested but with `findIndex`